apps/microfrontend-platform-devtools/src/app/intention-list/intention-list.component.ts

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (c) 2018-2020 Swiss Federal Railways
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+import { AfterViewInit, ChangeDetectionStrategy, Component, ElementRef, Input, OnChanges, Renderer2, SimpleChanges, ViewChild } from '@angular/core';
+import { Application, Intention } from '@scion/microfrontend-platform';
+import { Router } from '@angular/router';
+import { SciFilterFieldComponent } from '@scion/toolkit.internal/widgets';
+import { BehaviorSubject, combineLatest, Observable, of, Subject } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { filterIntentions, filterIntentionsFn, INTENTION_FILTER_PLACEHOLDER, setScionFilterFieldPlaceholder, splitFilter } from '../filter-utils';
+
+@Component({
+  selector: 'devtools-intention-list',
+  templateUrl: './intention-list.component.html',
+  styleUrls: ['./intention-list.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class IntentionListComponent implements OnChanges, AfterViewInit {
+
+  @Input()
+  public appIntentionMap: Map<Application, Intention[]>;
+  public applications$: Observable<Application[]>;
+
+  @ViewChild(SciFilterFieldComponent)
+  private _filterField: SciFilterFieldComponent;
+  @ViewChild(SciFilterFieldComponent, {read: ElementRef})
+  private _filterFieldElRef: ElementRef;
+  private _applications$ = new Subject<Application[]>();
+  private _filter$ = new BehaviorSubject<string[]>([]);
+
+  constructor(private _router: Router, private _renderer: Renderer2) {
+    this.installApplicationListener();
+  }
+
+  private installApplicationListener(): void {
+    this.applications$ = combineLatest([
```

Comment:
Right now there are five places where we load the applications. We could reduce the loading of the applications by implementing your suggestion, but I don't think it will actually simplify the code.